src/test/java/org/apache/commons/io/input/TeeInputStreamTest.java

```diff
@@ -30,6 +32,52 @@
  */
 public class TeeInputStreamTest  {
 
+    private static class ExceptionOnCloseByteArrayInputStream extends ByteArrayInputStream {
```

Comment:
@robtimus ,

Instead of creating these one-off classes, you should reuse and/or possibly refactor `YellOnFlushAndCloseOutputStream`, `ClosedOutputStream`, or `CloseShieldOutputStream`.

May you please give that a try?

Gary

Reply:
These classes are based on similar classes in `TeeOutputStreamTest`. While I could change the code to use `YellOnFlushAndCloseOutputStream`, I don't think that `ClosedOutputStream` or `CloseShieldOutputStream` is feasible, as these don't give access to its state; the test becomes more difficult to read because the check on whether or not `close()` has been called would require a try-fail-catch statement.